### ISSUE TITLE
Conntrack table not flushed for UDP nodePort service when doing deployment rollout restart

### ISSUE THREAD

**Issue by ravens** [kubernetes/kubernetes#96174]
**What happened**: 
Performing a `kubectl rollout restart deployment python-server-deployment` on a UDP-based service using nodePort leads to traffic from another pod to service being interrupted. Flushing manually the conntrack entry is enabling the traffic to flow again. 

**What you expected to happen**:
Traffic to continue flowing after rollout. 

**How to reproduce it (as minimally and precisely as possible)**:
The following [gist](https://gist.github.com/ravens/28022443b85332dd193f8446f6216994) using some basic python UDP code is exhibiting the issue when the rollout command is executed. The issue is happening often (2 times over 3 I would say). 

**Anything else we need to know?**:
I notice the k8s [code](https://github.com/kubernetes/kubernetes/blob/1e11e4a2108024935ecfcb2912226cedeafd99df/pkg/util/conntrack/conntrack.go#L119) doing the flushing is using a conntrack parameter that seems to be ineffective in that particular usecase:
```
$ kubectl get svc | grep python
python-server-service                     NodePort       10.17.110.58    <none>            5060:32499/UDP               25m
$ sudo conntrack -L | grep 5060
udp      17 118 src=10.40.63.169 dst=10.17.110.58 sport=5060 dport=5060 src=10.40.212.210 dst=10.40.63.169 sport=5060 dport=5060 [ASSURED] mark=0 use=1
conntrack v1.4.5 (conntrack-tools): 3208 flow entries have been shown.
$ sudo conntrack -D -p udp --dport 5060 --dst-nat 10.17.110.58
conntrack v1.4.5 (conntrack-tools): 0 flow entries have been deleted.
```
However, the following altered command line works:
```
$ sudo conntrack -D -p udp --dport 5060 --dst 10.17.110.58
udp      17 116 src=10.40.63.169 dst=10.17.110.58 sport=5060 dport=5060 src=10.40.212.210 dst=10.40.63.169 sport=5060 dport=5060 [ASSURED] mark=0 use=1
conntrack v1.4.5 (conntrack-tools): 1 flow entries have been deleted.
```

**Environment**:
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0", GitCommit:"e19964183377d0ec2052d1f1fa930c4d7575bd50", GitTreeState:"clean", BuildDate:"2020-08-26T14:23:04Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration:
baremetal
- OS (e.g: `cat /etc/os-release`):
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
- Kernel (e.g. `uname -a`):
Linux falcon 5.4.0-52-generic #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
kubeadm v.1.19.3
- Network plugin and version (if this is a network-related bug):
calico docker.io/calico/cni:v3.16.3, deployed using calico-typha operator - no network policy or endpoint objects deployed.
- Others:
Bug originally found on a kops-based cluster with 1.17.9 with SIP based traffic on port 5060, we checked on a more recent baremetal 1.19 cluster deployment.

**Comment by ravens** [comment]
Proxy mode is iptables btw.

**Comment by aojea** [comment]
so @ravens , the whole logic to delete conntrack entries in kube-proxy happens here

https://github.com/kubernetes/kubernetes/blob/71fea80155e5c124b1a44315da59af9799e9fafc/pkg/proxy/iptables/proxier.go#L1635-L1642

The code that you link has an interesting comment 

https://github.com/kubernetes/kubernetes/blob/1e11e4a2108024935ecfcb2912226cedeafd99df/pkg/util/conntrack/conntrack.go#L111-L119

that seems to link to an old issue with,  _the same problem you are reporting???_ https://github.com/kubernetes/kubernetes/issues/59368

I think that the problem is here
https://github.com/kubernetes/kubernetes/blob/6a3fd8f3d71e756f4231b610fb1d404abb100119/pkg/proxy/iptables/proxier.go#L769-L773

if is a nodePort it only deletes the NAT entries that are directed to the NodePort, but as we can see in your example, we are not deleting the entries directed to the ClusterIP, that as you demonstrated 
> sudo conntrack -D -p udp --dport 5060 --dst 10.17.110.58
it seems to work.

I think that removing the else so we ALWAYS delete the cluster IP -> endpoint conntrack entries will solve the problem
```
if nodePort != 0 {
   err = conntrack.ClearEntriesForPortNAT(proxier.exec, endpointIP, nodePort, svcProto)
} 
err = conntrack.ClearEntriesForNAT(proxier.exec, svcInfo.ClusterIP().String(), endpointIP, svcProto)

```

@ravens are you interested in take on this and fix it, and correct me if I'm wrong?

**Comment by ravens** [comment]
@aojea tentative PR #96251

**Comment by axot** [comment]
We are looking out a workaround before this fix that will be applied in 1.20, first trying is to decrease timeout related parameters as the following,

```
net.netfilter.nf_conntrack_udp_timeout
net.netfilter.nf_conntrack_udp_timeout_stream
```

But the side-effect is obviously, it will effect all nodePort type udp services.
